socMapping/StockSOC_ExtractPoints.py

```diff
@@ -0,0 +1,328 @@
+
+# Function make the server-side feature collection accessible to the client
+def getValues(fc):
+    features = fc.getInfo()['features']
+    dictarr = []
+    for f in features:
+        attr = f['properties']
+        dictarr.append(attr)
+    return dictarr
+
+
+# In[11]:
+
+
+# Convert input boundary Shapefile to a GEE boundary feature to constrain spatial extent
+boundary_ee = geemap.shp_to_ee(boundaryShp)
+
+
+# In[12]:
+
+
+# Get image data using temporal and spatial constraints
+s2_sr_cld_col = get_s2_sr_cld_col(boundary_ee, startDate, endDate)
+
+
+# In[13]:
+
+
+# Apply cloud/shadow mask and add NDVI layer
+sentinelCollection = (s2_sr_cld_col.map(add_cld_shdw_mask)
+                             .map(apply_cld_shdw_mask))
+
+
+# In[14]:
+
+
+# Create a list of dates for all images in the collection
+datesObject = sentinelCollection.aggregate_array("system:time_start")
+dateList =  datesObject.getInfo()
+dateList=[datetime.fromtimestamp(x/1000).strftime('%Y_%m_%d') for x in dateList]
+
+
+# In[15]:
+
+
+# Image display parameters
+sentinel_vis = {
+    'min': 0,
+    'max': 2500,
+    'gamma': [1.1],
+    'bands': ['B4', 'B3', 'B2']}
+
+
+# In[16]:
+
+
+# Convert input sample points Shapefile to a GEE feature
+points_ee = geemap.shp_to_ee(inPoints)
+
+
+# In[17]:
+
+
+# Dictionary to store all points
+allPoints = {}
+
+
+# In[18]:
+
+
+# Calculate Topographic wetness index and extract points
+upslopeArea = (ee.Image("MERIT/Hydro/v1_0_1")
+    .select('upa'))
+elv = (ee.Image("MERIT/Hydro/v1_0_1")
+    .select('elv'))
+
+slope = ee.Terrain.slope(elv)
+upslopeArea = upslopeArea.multiply(1000000).rename('UpslopeArea')
+slopeRad = slope.divide(180).multiply(math.pi)
+TWI = ee.Image.log(upslopeArea.divide(slopeRad.tan())).rename('TWI')
+extractedPointsTWI = geemap.extract_values_to_points(points_ee, TWI, scale=pixScale)
+dictarrTWI = getValues(extractedPointsTWI)
+
+
+# In[19]:
+
+
+# Read in and extract points for continuous heat-insolation load index and extract points
+chili = (ee.Image("CSP/ERGo/1_0/Global/SRTM_CHILI"))
+extractedPointsCHILI = geemap.extract_values_to_points(points_ee, chili, scale=pixScale)
+dictarrCHILI = getValues(extractedPointsCHILI)
+
+
+# In[20]:
+
+
+# Create a list of the images for processing
+images = sentinelCollection.toList(sentinelCollection.size())
+
+
+# In[21]:
+
+
+Map=geemap.Map()
+Map.centerObject(boundary_ee, 13)
+for index in range(0, sentinelCollection.size().getInfo()-1):
+    print("Processing " + dateList[index] + ": " + str(sentinelCollection.size().getInfo()-1 - index - 1) + " images to go      ", end = "\r")
+    image = ee.Image(images.get(index))
+    extractedPoints = geemap.extract_values_to_points(points_ee, image, scale=pixScale)
+    dictarr = getValues(extractedPoints)
+    points = gpd.GeoDataFrame(dictarr)
+    # Add the following variables to the collection of point data
+    points['stock'] = points['BD'] * points['C%']
+    points['twi'] = gpd.GeoDataFrame(dictarrTWI)['first']
```

Comment:
No change. I think this needs to be here since for each date I am creating an independent data set and both TWI and Chili need to be in each one. The processing was done outside of this loop and here I'm just pulling it into "points". At least that's what I think I'm doing.